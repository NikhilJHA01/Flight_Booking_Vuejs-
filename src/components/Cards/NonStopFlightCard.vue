<template>
  <section class="nonStopFlights">
    <el-card class="card" shadow="always">
      <div class="card__body">
        <div class="flex-column">
          <svg class="icon icon-aircraft-take-off">
            <use xlink:href="/sprite.svg#icon-aircraft-take-off"></use>
          </svg>
        </div>
        <div class="flex-column">
          <span class="primary--text">{{ nonStopFlight.name }}</span
          ><span class="secondary--text">{{ nonStopFlight.flightNo }}</span>
        </div>
        <div class="flex-column">
          <span class="primary--text">{{ nonStopFlight.departureTime }}</span
          ><span class="secondary--text">{{
            nonStopFlight.origin | removeCityCode
          }}</span>
        </div>
        <div class="flex-column">
          <span class="primary--text">{{ nonStopFlight.arrivalTime }}</span
          ><span class="secondary--text">{{
            nonStopFlight.destination | removeCityCode
          }}</span>
        </div>
        <div class="flex-column">
          <span class="primary--text">{{ nonStopFlight.totalHours }}</span
          ><span class="secondary--text">Non Stop</span>
        </div>
        <div class="flex-column">
          <span class="primary--text red-color">
            &#x20b9;{{ nonStopFlight.price * ticketCount }}</span
          >
          <span class="secondary--text">{{ ticketCount }} Ticket</span>
        </div>
        <el-button
          type="success"
          @click="bookingSuccess()"
          class="defaultBookButton"
          >Book</el-button
        >
      </div>
    </el-card>
  </section>
</template>

<script>
export default {
  name: "NonStopFlightCard",
  data() {
    return {};
  },
  props: {
    nonStopFlight: {
      type: Object,
      required: true
    },
    ticketCount: {
      type: Number,
      required: true
    }
  },
  components: {},
  methods: {
    // Emit event to Parent To Open the Booking dialog Message
    bookingSuccess() {
      this.$emit("showBookingMessage");
    }
  }
};
</script>

<style lang="scss" scoped>
.nonStopFlights {
  .card {
    margin: 10px;
    .icon {
      @extend %default-icon;
      stroke: $primary-main;
      fill: white;
    }
    &__body {
      @extend %default-card-body;
    }
  }
}
</style>
