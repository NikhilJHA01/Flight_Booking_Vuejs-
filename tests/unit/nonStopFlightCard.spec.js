import { mount } from "@vue/test-utils";
import NonStopFlightCard from "@/components/Cards/NonStopFlightCard";
import { oneWayFlightData } from "./data/flightTestData";

describe("NonStopFlightCard", () => {
  const wrapper = mount(NonStopFlightCard, {
    propsData: {
      nonStopFlight: oneWayFlightData,
      ticketCount: 1
    }
  });
  it("renders the Non Stop Card", async () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Fires a Book event on succesful book from Non-stop Flights", async () => {
    wrapper.find("el-button").trigger("click");
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted("showBookingMessage")).toBeTruthy();
  });

  it("check NonStop Flights card has been rendered and it contains svg", () => {
    const form = wrapper.find("svg");
    expect(form.exists()).toBe(true);
  });
});
