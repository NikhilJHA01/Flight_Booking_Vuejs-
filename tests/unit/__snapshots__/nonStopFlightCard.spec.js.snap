// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NonStopFlightCard renders the Non Stop Card 1`] = `
<section>
  <el-card shadow="always" class="card">
    <div class="card__body">
      <div class="flex-column"><svg class="icon icon-aircraft-take-off">
          <use xlink:href="/sprite.svg#icon-aircraft-take-off"></use>
        </svg></div>
      <div class="flex-column"><span class="primary--text">SpiceJet</span><span class="secondary--text">SJ-105</span></div>
      <div class="flex-column"><span class="primary--text">8:15</span><span class="secondary--text">Pune (PNQ)</span></div>
      <div class="flex-column"><span class="primary--text">9:15</span><span class="secondary--text">Bengaluru (BLR)</span></div>
      <div class="flex-column"><span class="primary--text"></span><span class="secondary--text">Non Stop</span></div>
      <div class="flex-column"><span class="primary--text red-color">
          â‚¹3549</span> <span class="secondary--text">1 Ticket</span></div>
      <el-button type="success" round="">Book</el-button>
    </div>
  </el-card>
</section>
`;
